homeassistant/components/emulated_hue.py

```diff
@@ -0,0 +1,538 @@
+
+        if entity.domain.lower() in config.off_maps_to_on_domains:
+            # Map the off command to on
+            service = SERVICE_TURN_ON
+
+            # Caching is required because things like scripts and scenes won't
+            # report as "off" to Alexa if an "off" command is received, because
+            # they'll map to "on". Thus, instead of reporting its actual
+            # status, we report what Alexa will want to see, which is the same
+            # as the actual requested command.
+            self.cached_states[entity_id] = (result, brightness)
+
+        # Perform the requested action
+        self.hass.services.call(core.DOMAIN, service, data, blocking=True)
+
+        json_response = [create_hue_success_response(entity_id, 'on', result)]
+
+        if brightness is not None:
+            json_response.append(
+                create_hue_success_response(entity_id, 'bri', brightness))
+
+        return self.json(json_response)
+
+    def is_entity_exposed(self, entity):
+        """Determine if an entity should be exposed on the emulated bridge."""
+        config = self.config
+
+        if entity.attributes.get('view') is not None:
+            # Ignore entities that are views
+            return False
+
+        domain = entity.domain.lower()
+        explicit_expose = entity.attributes.get(ATTR_EMULATED_HUE, None)
+
+        domain_exposed_by_default = \
+            config.expose_by_default and domain in config.exposed_domains
+
+        # Expose an entity if the entity's domain is exposed by default and
+        # the configuration doesn't explicitly exclude it from being
+        # exposed, or if the entity is explicitly exposed
+        is_default_exposed = \
+            domain_exposed_by_default and explicit_expose is not False
+
+        return is_default_exposed or explicit_expose
+
+
+def parse_hue_api_put_light_body(request_json, entity):
+    """Parse the body of a request to change the state of a light."""
+    if 'on' in request_json:
+        if not isinstance(request_json['on'], bool):
+            return None
+
+        if request_json['on']:
+            # Echo requested device be turned on
+            brightness = None
+            report_brightness = False
+            result = True
+        else:
+            # Echo requested device be turned off
+            brightness = None
+            report_brightness = False
+            result = False
+
+    if 'bri' in request_json:
+        # Make sure the entity actually supports brightness
+        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+        if (entity_features & SUPPORT_BRIGHTNESS) == SUPPORT_BRIGHTNESS:
+            try:
+                # Clamp brightness from 0 to 255
+                brightness = max(0, min(int(request_json['bri']), 255))
+            except ValueError:
+                return None
+
+            report_brightness = True
+            result = True if brightness > 0 else False

```

Comment:
Oh, I see. Sorry, wasn't able to get that from your original comment. I'll fix with the other issues.